### PR TITLE
`sage.graphs.generic_graph`: some care with return ... else statements

### PR DESCRIPTION
**Pull request by dcoudert**
We change this kind of statements:
```py
if ....:
    return ...
else:
    return ...
```
to
```py
if ....:
    return ...
return ...
```

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/graphs/generic_graph.py [hunk 1]
```diff
@@ -1709,9 +1709,20 @@ def igraph_graph(self, vertex_list=None, vertex_attrs={}, edge_attrs={}):
...
             sage: G.edges(sort=True) == H.edges(sort=True)             # optional - python_igraph
             False
+
+        Check input parameter ``vertex_list``::
+
+            sage: G = Graph(3)
+            sage: G.igraph_graph(vertex_list=[])                       # optional - python_igraph
+            Traceback (most recent call last):
+            ...
+            ValueError: ``vertex_list`` must be a permutation of the vertices
```
Comment (fchapoton) [inline src/sage/graphs/generic_graph.py]: please no rst syntax in error messages

**Comment by fchapoton** [comment]
otherwise, looks good, though I am not convinced about the priority to give to such changes

File: src/sage/graphs/generic_graph.py (hunk 1 above)
Comment (dcoudert) [reply src/sage/graphs/generic_graph.py]: I have removed rst syntax from many errors messages.

**Review by fchapoton — APPROVED** [review]
ok, let's go

**Comment by dcoudert** [comment]
Thank you.

**Pull request merged**